:bomb: `Bug`: jenkins-plugin failed

### What Happened?

tools file:

```yaml
tools:
  - name: jenkins
    instanceID: default
    options:
      test_env: true
      create_namespace: true
      repo:
        name: jenkins
        url: https://charts.jenkins.io
      chart:
        chart_name: jenkins/jenkins
        release_name: jenkins
        namespace: jenkins
        wait: true
        timeout: 10m
        upgradeCRDs: true
        values_yaml: |
          persistence:
            storageClass: jenkins-pv
          serviceAccount:
            create: false
            name: jenkins
          controller:
            serviceType: NodePort
            nodePort: 32000
            additionalPlugins:
              # install "GitHub Pull Request Builder" plugin, see https://plugins.jenkins.io/ghprb/ for more details
              - ghprb
              # https://plugins.jenkins.io/gitlab-plugin/
              - gitlab-plugin
              - antisamy-markup-formatter
            # Enable HTML parsing using OWASP Markup Formatter Plugin (antisamy-markup-formatter), useful with ghprb plugin.
            enableRawHtmlMarkupFormatter: true
            # Jenkins Configuraction as Code, refer to https://plugins.jenkins.io/configuration-as-code/ for more details
            # notice: All configuration files that are discovered MUST be supplementary. They cannot overwrite each other's configuration values. This creates a conflict and raises a ConfiguratorException.
            JCasC:
              defaultConfig: true
              # each key-value in configScripts will be added to the ${JENKINS_HOME}/casc_configs/ directory as a file.
              configScripts:
                # this will create a file named "safe_html.yaml" in the ${JENKINS_HOME}/casc_configs/ directory.
                # it is used to configure the "Safe HTML" plugin.
                # filename must meet RFC 1123, see https://tools.ietf.org/html/rfc1123 for more details
```

Errors:

```shell
2022/08/18 13:41:59 release installed successfully: jenkins/jenkins-4.1.17
2022-08-18 13:41:59 ℹ [INFO]  Here is how to get the password of the admin user:
2022-08-18 13:41:59 ℹ [INFO]  kubectl exec --namespace jenkins -it svc/jenkins-jenkins -c jenkins -- /bin/cat /run/secrets/additional/chart-admin-password && echo
2022-08-18 13:41:59 ℹ [INFO]  You are in test env. Here are the Jenkins url in host machine and Jenkins url in K8s cluster.
2022-08-18 13:41:59 !! [ERROR]  Failed to get jenkins url: exit status 1
2022/08/18 13:42:00 WARNING: repository name "jenkins" already exists
2022-08-18 13:42:00 ℹ [INFO]  Uninstalling jenkins helm chart.
2022/08/18 13:42:00 uninstall: Deleting jenkins
2022/08/18 13:42:01 Starting delete for "jenkins" Service
2022/08/18 13:42:01 Starting delete for "jenkins-agent" Service
2022/08/18 13:42:01 Starting delete for "jenkins" StatefulSet
2022/08/18 13:42:01 Starting delete for "jenkins-watch-configmaps" RoleBinding
2022/08/18 13:42:01 Starting delete for "jenkins-schedule-agents" RoleBinding
2022/08/18 13:42:01 Starting delete for "jenkins-casc-reload" Role
2022/08/18 13:42:01 Starting delete for "jenkins-schedule-agents" Role
2022/08/18 13:42:01 Starting delete for "jenkins" PersistentVolumeClaim
2022/08/18 13:42:01 Starting delete for "jenkins-jenkins-jcasc-config" ConfigMap
2022/08/18 13:42:01 Starting delete for "jenkins" ConfigMap
2022/08/18 13:42:01 Starting delete for "jenkins" Secret
2022/08/18 13:42:01 purge requested for jenkins
2022/08/18 13:42:01 release uninstalled, response: &{0x1400048b7a0 }
2022-08-18 13:42:01 ℹ [INFO]  -------------------- [  Processing aborted.  ] --------------------
2022-08-18 13:42:01 !! [ERROR]  Errors Map: key(jenkins/default-Create) -> value(Failed to get jenkins url: exit status 1)
2022-08-18 13:42:01 !! [ERROR]  Apply failed => some error(s) occurred during plugins apply process.
```

### How to Reproduce?

_No response_

### Anything else

_No response_

### DevStream Version

latest

--- a/internal/pkg/plugin/jenkins/options.go
+++ b/internal/pkg/plugin/jenkins/options.go
@@ -1,6 +1,9 @@
 package jenkins
 
 import (
+	"fmt"
+	"strings"
+
 	"github.com/mitchellh/mapstructure"
 
 	"github.com/devstream-io/devstream/internal/pkg/plugininstaller"
@@ -38,4 +41,13 @@ func setDefaultValue(defaultOpts *helm.Options) plugininstaller.MutableOperation
 		opts.FillDefaultValue(defaultOpts)
 		return opts.encode()
 	}
+}
+
+// refer: https://github.com/jenkinsci/helm-charts/blob/30766b45faf639dbad45e2c66330ee5fcdc7e37f/charts/jenkins/templates/_helpers.tpl#L46-L57
+func (opts *jenkinsOptions) getJenkinsFullName() string {
+	if strings.Contains(opts.Chart.ChartName, opts.Chart.ReleaseName) {
+		return opts.Chart.ReleaseName
+	}
+
+	return fmt.Sprintf("%s-%s", opts.Chart.ReleaseName, opts.Chart.ChartName)
 }

--- a/internal/pkg/plugin/jenkins/prompt.go
+++ b/internal/pkg/plugin/jenkins/prompt.go
@@ -10,10 +10,11 @@ import (
 )
 
 func buildPasswdOfAdminCommand(opts jenkinsOptions) string {
-	method := fmt.Sprintf("kubectl exec --namespace jenkins -it svc/%s-jenkins -c jenkins "+
-		"-- /bin/cat /run/secrets/additional/chart-admin-password && echo", opts.Chart.ReleaseName)
+	jenkinsFullName := opts.getJenkinsFullName()
 
-	return method
+	return fmt.Sprintf("kubectl exec --namespace %s -it svc/%s -c jenkins "+
+		"-- /bin/cat /run/secrets/additional/chart-admin-password && echo", opts.Chart.Namespace, jenkinsFullName)
+
 }
 
 func howToGetPasswdOfAdmin(options plugininstaller.RawOptions) error {
@@ -48,21 +49,25 @@ func getPasswdOfAdmin(options plugininstaller.RawOptions) (string, error) {
 }
 
 // getJenkinsURL returns the jenkins url of the jenkins, format: hostname:port
+// there are duplicated code in getJenkinsURLForTestEnv, need to refactor
 func getJenkinsURL(options plugininstaller.RawOptions) (string, error) {
 	opts, err := newOptions(options)
 	if err != nil {
 		return "", err
 	}
 
+	jenkinsFullName := opts.getJenkinsFullName()
 	commands := []string{
 		`jsonpath="{.spec.ports[0].nodePort}"`,
-		fmt.Sprintf(`NODE_PORT=$(kubectl get -n jenkins -o jsonpath=$jsonpath services %s-jenkins)`, opts.Chart.ReleaseName),
+		fmt.Sprintf(`NODE_PORT=$(kubectl get -n %s -o jsonpath=$jsonpath services %s)`, opts.Chart.Namespace, jenkinsFullName),
 		`jsonpath="{.items[0].status.addresses[0].address}"`,
-		`NODE_IP=$(kubectl get nodes -n jenkins -o jsonpath=$jsonpath)`,
+		fmt.Sprintf(`NODE_IP=$(kubectl get nodes -n %s -o jsonpath=$jsonpath)`, opts.Chart.Namespace),
 		`echo $NODE_IP:$NODE_PORT`,
 	}
+	commandsOneLine := strings.Join(commands, " && ")
+	log.Debugf("commands of get jenkins url: %s", commandsOneLine)
 
-	cmd := exec.Command("sh", "-c", strings.Join(commands, " && "))
+	cmd := exec.Command("sh", "-c", commandsOneLine)
 
 	output, err := cmd.Output()
 	if err != nil {
@@ -79,15 +84,18 @@ func getJenkinsURLForTestEnv(options plugininstaller.RawOptions) (string, error)
 		return "", err
 	}
 
+	jenkinsFullName := opts.getJenkinsFullName()
 	commands := []string{
 		`jsonpath="{.spec.ports[0].nodePort}"`,
-		fmt.Sprintf(`NODE_PORT=$(kubectl get -n jenkins -o jsonpath=$jsonpath services %s-jenkins)`, opts.Chart.ReleaseName),
+		fmt.Sprintf(`NODE_PORT=$(kubectl get -n %s -o jsonpath=$jsonpath services %s)`, opts.Chart.Namespace, jenkinsFullName),
 		`jsonpath="{.items[0].status.addresses[0].address}"`,
 		`NODE_IP=127.0.0.1`,
 		`echo $NODE_IP:$NODE_PORT`,
 	}
+	commandsOneLine := strings.Join(commands, " && ")
+	log.Debugf("commands of get jenkins url: %s", commandsOneLine)
 
-	cmd := exec.Command("sh", "-c", strings.Join(commands, " && "))
+	cmd := exec.Command("sh", "-c", commandsOneLine)
 
 	output, err := cmd.Output()
 	if err != nil {
@@ -130,7 +138,7 @@ func showJenkinsUrl(options plugininstaller.RawOptions) error {
 			log.Error(err)
 			return err
 		}
-		log.Info("Jenkins url in K8s:", fmt.Sprintf("http://%s/login", urlInK8s))
+		log.Info("Jenkins url in K8s: ", fmt.Sprintf("http://%s/login", urlInK8s))
 
 	}
 

